uniforce/lib/Cleantalk/USP/Security/Firewall.php

```diff
@@ -166,43 +166,58 @@ public function run() {
 	 * @return array Single element array of result
 	 */
 	private function prioritize( $results ){
-		
-		$current_fw_result_priority = 0;
-		$result = array(
-			'status' => 'PASS',
-			'ip' => '',
-		);
-		
-		if( is_array( $results ) ) {
-			foreach ( $results as $fw_result ) {
-				$priority = array_search( $fw_result['status'], $this->statuses_priority ) + ( isset($fw_result['is_personal']) && $fw_result['is_personal'] ? count ( $this->statuses_priority ) : 0 );
-				if( $priority >= $current_fw_result_priority ){
-					$current_fw_result_priority = $priority;
+
+        $final_priority = 0;
+        $result = array(
+            'status' => 'PASS',
+            'ip' => '',
```

Comment:
If the array does not come to the parameter, then the result with an empty value in the IP field is suitable for us?

Reply:
Yes, if no module results are gained we just pass this request through. Note, this logic is used twice, for each module and for the conjunction of modules results.